### PR TITLE
Initial Project Structure PR

### PR DESCRIPTION
**Pull request by domasmas**
Testing pull requests

### CODE REVIEW

File: source/trunk/Esf.DataAccess.Tests/Esf.DataAccess.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,88 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.DataAccess.Tests/Esf.DataAccess.Tests.csproj]: I think we should group together test projects with the functionality projects: it: in folder Esf.DataAccess projects Esf.DataAccess and Esf.DataAccess.Tests

File: source/trunk/Esf.Web/App_Start/RouteConfig.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.Web.Mvc;
+using System.Web.Routing;
+
+namespace Esf.Web
```
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.Web/App_Start/RouteConfig.cs]: I would name the website project Esf.Website or something need to reflects that this is the website + frontend stuff like typescript css, etc.

File: source/trunk/Esf.Web/Controllers/HomeController.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Web.Mvc;
+
+namespace Esf.Web.Controllers
+{
+    public class HomeController : Controller
```
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.Web/Controllers/HomeController.cs]: Are we using the latest project templates for ASP .NET MVC 4.5?

File: source/trunk/Esf.WebApi/Esf.WebApi.csproj [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+    <Content Include="Web.Debug.config">
+      <DependentUpon>Web.config</DependentUpon>
+    </Content>
+    <Content Include="Web.Release.config">
+      <DependentUpon>Web.config</DependentUpon>
+    </Content>
+  </ItemGroup>
+  <ItemGroup>
+    <Folder Include="App_Data\" />
+    <Folder Include="Areas\" />
```
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.WebApi/Esf.WebApi.csproj]: why do we need Areas?

File: source/trunk/Esf.WebApi/Esf.WebApi.csproj [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+    <Folder Include="Content\" />
```
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.WebApi/Esf.WebApi.csproj]: We don't need Content, Controllers, fonts, models, Results and Scripts.

File: source/trunk/Esf.WebApi/Esf.WebApi.csproj [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+    <Folder Include="Controllers\" />
+    <Folder Include="fonts\" />
+    <Folder Include="Models\" />
+    <Folder Include="Providers\" />
```
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.WebApi/Esf.WebApi.csproj]: Why do we need Providers?

File: source/trunk/Esf.WebApi/Esf.WebApi.csproj [hunk 7]
```diff
@@ -0,0 +1,276 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.WebApi/Esf.WebApi.csproj]: Esf.WebApi is good name as log as we don't associate it with the concrete microsoft technology. If we use this I'll like to use it as an internal term for Http Service or other cross machine service.

File: source/trunk/Esf.sln [hunk 8]
```diff
@@ -0,0 +1,65 @@
+﻿
```
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.sln]: why do we have source and nested folder trunk? I would put the whole code in source and remove the prototype from the main source folder. Also we can keep prototypes in different branches outside the main source.

File: source/trunk/Esf.sln [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.sln]: I think it's good idea to redirect the output of all the projects(the bin folders) to one deployment directory.

File: source/trunk/Esf.sln [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+# Visual Studio 14
```
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.sln]: I can't build the concept project.

Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.sln]: It's good to have separate repositories for big resources - like dlls, third party dependencies that we need to version as they will evolve slow compared to our code base.
Not sure if this is good idea for the plannig documents. The bin folder of the concept project shouldn't be in source control as this is output. We can keep it in separate snapshot(drop) repository.

Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.sln]: I thought you would be integrating the concept project directly into Esf.Web. Is this a separate step we'll do later or this is just a throw-away prototype?

File: source/trunk/Esf.Web/App_Start/RouteConfig.cs (hunk 2 above)
Comment (vtrifonov-esfiddle) [inline source/trunk/Esf.Web/App_Start/RouteConfig.cs]: I guess we'll setup less, typescript and the website folder structure as a separate task? It must be definitely a separate checkin atleast.

File: source/trunk/Esf.DataAccess.Tests/Esf.DataAccess.Tests.csproj (hunk 1 above)
Comment (domasmas) [inline source/trunk/Esf.DataAccess.Tests/Esf.DataAccess.Tests.csproj]: Agree

File: source/trunk/Esf.WebApi/Esf.WebApi.csproj (hunk 4 above)
Comment (domasmas) [inline source/trunk/Esf.WebApi/Esf.WebApi.csproj]: Areas are needed to separate controllers and views by feature

File: source/trunk/Esf.WebApi/Esf.WebApi.csproj (hunk 5 above)
Comment (domasmas) [inline source/trunk/Esf.WebApi/Esf.WebApi.csproj]: Controllers will be put inside areas.

File: source/trunk/Esf.WebApi/Esf.WebApi.csproj (hunk 7 above)
Comment (domasmas) [inline source/trunk/Esf.WebApi/Esf.WebApi.csproj]: Ok, Esf.WebApi is generic enough, we'll continue to use it.

File: source/trunk/Esf.sln (hunk 8 above)
Comment (domasmas) [inline source/trunk/Esf.sln]: Create a separate folder for prototypes. Remove trunk folder.

File: source/trunk/Esf.sln (hunk 9 above)
Comment (domasmas) [inline source/trunk/Esf.sln]: Build content should go to /deployment/Website/, /deployment/WebApi/, etc.

**Pull request closed without merging**